### ISSUE TITLE
Set up localization

### ISSUE THREAD

**Issue by secretrobotron** [mozilla/id.webmaker.org#10]
- [ ] Create a new transifex group/project? ( @alicoding ?? )
- [ ] Sync pages (#6, #7, #8) with transifex
- [ ] Send out alert to transifex community to ask for help

**Comment by alicoding** [comment]
If this is replacing the old login I think we can simply remove the old file from Transifex and just replace the new one instead. As long as we don't have way too many strings for this project otherwise we should setup its own project on Transifex.

**Comment by secretrobotron** [comment]
:+1:
Since we're initially just servicing teach.mozilla.org, we're going to have to do this in stages though. But, maybe we should just "shut down" localization changes to current login system in anticipation of it going away.

**Comment by alicoding** [comment]
Then lets do this:

1. Stop accepting new translation on old login
2. Upload new translation file (whatever format we are going to use for this new login. PO file or JSON whatever we are going to use)
3. Once we have everything in place then remove the old one and just keep the new one in the same project if it is not too heavy to be part of the existing Transifex project.

**Comment by secretrobotron** [comment]
Cool. The pages should be fairly light-weight. I don't think we're going to tackle localization of transactional emails here yet.

What do you recommend for the translation format? Can you help @jbuck @cadecairos and @Pomax tinker with framework decisions? Don't see why we shouldn't use your existing i18n stuff.

**Comment by alicoding** [comment]
We don't have localization ready for React, yes. I'm going to drop this from this milestone for now.

**Comment by simonwex** [comment]
Does this need to be in this milestone? l10n of the teach site isn't a requirement for launch.

**Comment by thisandagain** [comment]
To the backlog!

**Comment by cadecairos** [comment]
@gideonthomas @alicoding @hannahkane @simonwex

**Comment by alicoding** [comment]
A bit more details here... learning.mozilla.org is being localized and one of the flow (login obviously) is not localized and for the new login to be implemented it seems like going to take a while until that happened. We need to decide whether we want to localize id.wb.org now or wait until the new one is implemented. Going to give pro&cons to do this now without waiting.

Pros:

1. We have fully localized app for learning site with the full flow.

Cons:

1. It's a pain to localize this old app because it uses very old technology, but I have done it so not too hard
2. People will have to translate something that will be removed in the near future...

**Comment by cadecairos** [comment]
> in the near future...

keep in mind we don't even have a timeline for deprecating this service yet

**Comment by hannahkane** [comment]
Just want to note that the only thing on the Learning site now that requires authentication is applying to be a Club Captain. We haven't yet agreed upon a localization strategy for that form/workflow yet (see https://github.com/mozilla/learning.mozilla.org/issues/2049).

**Comment by alicoding** [comment]
@hannahkane but it's not the only thing that's being used for in that site, right? We used to also show them all their projects too? The reason I ask or want to confirm this because for the Club Captain it might not be that important since we do required that they understand English because that's how we curate their applications, right?

**Comment by Pomax** [comment]
correct, we still have the "your projects" section that is tied to id.wmo login, and if we flip the badges feature flag, that too relies on id.wmo login.

**Comment by hannahkane** [comment]
oh, shoot, forgot about those things. I'm leaning towards localizing then.

But, yes, @alicoding, there's a discussion happening in that ticket about how we can evaluate/curate applications in multiple languages. For now, it's EN only.

**Comment by gideonthomas** [comment]
@simonwex or @hannahkane, is there a status update on this? Last I remember, this was not going to happen. If so, can we close this issue?

**Comment by hannahkane** [comment]
Do we have an effort estimate for this?

**Comment by alicoding** [comment]
This project is using an older version of react (0.12.2) which is about 3 major version behind (current version is 15.3.2, don't mind the 0&15 here)... If we just want to get it localized without making it perfect then this is not too hard to get setup and localized since we have done it before for donate repo and the API should be similar if you look at the older version here...

**Comment by hannahkane** [comment]
Hi @alicoding!

Without knowing what this is competing against, I'm finding it hard to help prioritize. @gideonthomas or @cadecairos - can you provide a list of things we might do instead of this? Then I could offer my two cents on where this fits in the list. Or maybe @simonwex has a more concrete answer.

**Comment by gideonthomas** [comment]
Yeah I think @simonwex might know. It all depends on what our strategy for authentication is going to be in the coming year and/or if we're switching away from id.wm.org.
